mindsdb/utilities/context_executor.py

```diff
@@ -57,3 +60,49 @@ def execute_in_threads(func, tasks, thread_count=3, queue_size_k=1.5):
 
         time.sleep(0.1)
     executor.shutdown(wait=False)
+
+
+def process_dataframe_in_partitions(df, callback, partition_size):
+    """
+    Splits dataframe into partitions and apply callback on each partition
+
+    :param df: input dataframe
+    :param callback: function to apply on each partition
+    :param partition_size: size of each partition
+    :return: yield results
+    """
+
+    # tasks
+    def split_data_f(df):
+        chunk = 0
+        while chunk * partition_size < len(df):
+            # create results with partition
+            df1 = df.iloc[chunk * partition_size: (chunk + 1) * partition_size]
+            chunk += 1
+            yield [df1]
+
+    tasks = split_data_f(df)
+
+    # workers count
+    is_cloud = Config().get('cloud', False)
+    if is_cloud:
+        max_threads = int(os.getenv('MAX_QUERY_PARTITIONS', 10))
```

Comment:
will be better to get this value from `Config`

Reply:
The name `MAX_QUERY_PARTITIONS` is not relevant. It is not max partitions count.

Reply:
It is max number of partitions to split execution, what is the better name, MAX_PARTITIONS_COUNT?

Reply:
`max number of partitions to split execution` sounds like chunk_count from line 94. It is someting like `MAX_QUERY_PROCESSING_STREAMS_COUNS`, or `DATAFRAME_PARALLEL_PROCESSING_MAX_THREADS_COUNT`, or any other cool long name.

Reply:
and of cource with `MINDSDB_%` prefix :)

Reply:
renamed to MINDSDB_MAX_PARTITIONING_THREADS